agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt

```diff
@@ -37,6 +37,18 @@ class OfflineAgendaRepository(
         }
     }
 
+    override fun fetchAgendaItemTask(id: String): Flow<TaskyResult<AgendaItem.Task>> {
+        return networkApiCall {
+            taskyRetrofitApi.getTask(id).toAgendaTaskItem()
+        }
+    }
+
+    override fun fetchAgendaItemReminder(id: String): Flow<TaskyResult<AgendaItem.Reminder>> {
+        return networkApiCall {
+            taskyRetrofitApi.getReminder(id).toAgendaReminderItem()
+        }
+    }
+
```

Comment:
`Flow`s make more sense when you expect a continuous flow of values that need to be observed (like observing changes in user location updates or changes in a database table). In these cases a normal `suspend` function makes more sense when returning a single result.